ls1intum/ArTEMiS
# Improve course exercise details page

### Checklist

### Description


When testing the result display in the course exercise detail component in https://github.com/ls1intum/ArTEMiS/pull/528 in combination with the code-editor, I noticed two issues:

- **The latest result is not correct**, it is always the first graded result in list.
=> Take the most recent graded result

- **The result list is not complete when going back and forth from the code-editor** (potentially also other pages). The reason for this is that the code-editor only loads the most recent result, attaches it to the participation and puts this object intot the participation websocket service storage. When the exercise detail page subscribes to it, it just uses this object that is missing the results.
=> We changed the participation websocket service so that one can subscribe for participation results without providing a participation object. This makes sense for components that are only interested in the most recent result.

- ~**Quiz state was not updated on submit**. The problem was that the QuizScheduleService only sends a new participation object by WS to the client when the quiz has ended - however we need it also after the submission as this changes the state on the detail page. Because of this issue sometimes you still saw "start exercise" instead of "view submission" on the detail page for quizzes.
Note: As this task is done by the QuizScheduler, there can still be a small delay until the start exercise button is replaced with view submission.~
This fix lead to a lot of new bugs. I created a ticket for this issue.

- **We show non-graded results in the "Recent results" section** (under "Your graded results" we still only show graded results or "You have no graded result.")
- **We hide the result history bar if only one result exists**

See more info about the last two changes in https://github.com/ls1intum/ArTEMiS/pull/528.

### Steps for Testing


1. Log in to ArTEMiS
2. Go to the exercise detail page of a programming exercise
3. Open the code-editor, submit a couple of times, make the last submit a build error
5. Reload the page
6. Go back to exercise-detail
=> All results should show
=> Latest result should be correct
